app/models/article.rb

```diff
@@ -565,6 +566,12 @@ def past_or_present_date
     end
   end
 
+  def canonical_url_must_not_has_spaces
+    if canonical_url.to_s.match(/\s/).present?
```

Comment:
The issue I see with this regexp is that it's not going to match the [other Unicode spaces](https://www.compart.com/en/unicode/category/Zs).

This is a regular space, as you can see it's matched:

![Screenshot_2020-02-17 Rubular(1)](https://user-images.githubusercontent.com/146201/74614553-f7ff7380-5118-11ea-99e1-000e1a2db827.png)

but this example is a unicode whitespace character and as you can see, it's not going to be matched, so the URL will pass validation, but be still likely broken:

![Screenshot_2020-02-17 Rubular](https://user-images.githubusercontent.com/146201/74614573-3006b680-5119-11ea-855e-dfc8aacfdde1.png)

My suggestion is to use `[[:space:]]` instead of `\s` which will match all whitespace chars.

You can also add a test with one of those spaces instead of the regular space to check it works :)

Reply:
@rhymes I’ll change it and yes I’ll add test for it. Thanks!

Reply:
I didn't Unicode whitespace test, I don't know how to write Unicode char in test file but it covers Unicode chars